girder/stemserver_plugin/stemimage.py

```diff
@@ -21,6 +21,8 @@ def __init__(self):
         self.route('GET', (':id', 'dark'), self.dark)
         self.route('GET', (':id', 'bright', 'shape'), self.bright_shape)
         self.route('GET', (':id', 'dark', 'shape'), self.dark_shape)
+        self.route('GET', (':id', 'electron', 'frames'), self.electron_frames)
+        self.route('GET', (':id', 'electron', 'scans'), self.electron_scans)
```

Comment:
I think we want to make is a little more generic, so we could use the same endpoint to serve full diffractograms. So I think we should do some renaming. We also want to be able to access a single frame rather than have to pull whole set.  
`GET /stemimages/:id/frames/:scanposition` - Would return a single diffractogram for the given position
`GET /stemimages/:id/frames` - Would return all the diffractograms.
`GET /stemimages/:id/scan-positions` - Would return all the scan positions.

Reply:
@cjh1 I generalized the endpoint names as mentioned here. But currently, the functions have hard-wired within them the HDF5 paths to the electron counting diffractograms and scans ( `/electron_events/frames` and `/electron_events/scan_positions` ).

Maybe if we add more diffractograms in the future, we can add additional parameters/options to the endpoint so the user can specify which diffractogram they want to get?